src/humanpy/hum_kin1.py

```diff
@@ -0,0 +1,263 @@
+        vect_he = (sys_elbow[0:3,3] - sys_hand[0:3,3])/ \
+                  numpy.linalg.norm([sys_elbow[0:3,3] - sys_hand[0:3,3]])
+        vect_norm_es_eh = numpy.cross(vect_he, vect_es)
+        
+        q4 = - numpy.arccos(numpy.asscalar(numpy.dot(vect_he.T,vect_es)))                   #[0,-pi]
+        
+        if side == 'L': 
+            q2 = numpy.asscalar(numpy.arcsin(self.checkArg(-vect_es[0])))                   #[-pi/2,pi/2]
+        else:
+            q2 = numpy.asscalar(numpy.arcsin(self.checkArg(vect_es[0]))) 
+        if sys_sh_elb[1,3] > 0:                                                             #projection on y of the sys origin. if y positive, add pi/2
+            q2 = q2 + numpy.sign(q2)*numpy.pi/2                                             #[-pi,pi]
+
+        if numpy.abs(numpy.cos(q2)) > 0.1:
+            q1 = numpy.asscalar(numpy.arccos(self.checkArg(vect_es[1]/numpy.cos(q2))))      #[0,pi]
+            if numpy.asscalar(numpy.arcsin(self.checkArg(-vect_es[2]/numpy.cos(q2)))) < 0:  #[-pi,pi]
+                q1 = -q1
+            
+            q3 = numpy.asscalar(numpy.arccos(self.checkArg(-vect_norm_es_eh[0]/numpy.cos(q2))))                
+        else:
+            q1 = None
+            q3 = None
+
+        return [q1, q2, q3, q4]
+
+    def getLowerLimbAngles(self, tf, side):
+        if side == 'L':        
+            sys_hip = self.getSkeletonTransformation(tf, 'left_hip', KIN_FRAME)
+            sys_knee = self.getSkeletonTransformation(tf, 'left_knee', KIN_FRAME)
+            sys_foot = self.getSkeletonTransformation(tf, 'left_foot', KIN_FRAME)
+        else:            
+            sys_hip = self.getSkeletonTransformation(tf, 'right_hip', KIN_FRAME)
+            sys_knee = self.getSkeletonTransformation(tf, 'right_knee', KIN_FRAME)
+            sys_foot = self.getSkeletonTransformation(tf, 'right_foot', KIN_FRAME)
+            
+        if sys_hip is None or sys_knee is None or sys_foot is None:
+            return None
+        
+        vect_kh = (sys_hip[0:3,3] - sys_knee[0:3,3])/  \
+                  numpy.linalg.norm([sys_hip[0:3,3] - sys_knee[0:3,3]])
+        vect_fk = (sys_knee[0:3,3] - sys_foot[0:3,3])/ \
+                  numpy.linalg.norm([sys_knee[0:3,3] - sys_foot[0:3,3]])
+        q2 = - numpy.arccos(self.checkArg(numpy.asscalar(numpy.dot(vect_kh.T,vect_fk))))
+        
+        q1 = numpy.asscalar(numpy.arccos(vect_kh[1]))                                       #[0,pi]
+        if numpy.asscalar(numpy.arcsin(vect_kh[2])) < 0:                                    #[-pi,pi]
+            q1 = -q1 
+            
+        return [q1, q2]
+  
+    def checkLimits(self, joint, angle_vect, angle_elem, collection):
+        if angle_vect is not None:
+            angle = angle_vect[angle_elem]
+            if angle is not None and joint is not None:
+                lower,upper = joint.GetLimits()
+                if angle < lower: angle = lower
+                if angle > upper: angle = upper
```

Comment:
This clamping logic is duplicated several places. I'd prefer that you defined a helper function that looked something like this to eliminate the duplication:
```python
def constrain(x, x_min, x_max):
    return min(max(x, x_min), x_max)